pkg/deploy/lattice/service_network_synthesizer.go

```diff
@@ -117,7 +81,39 @@ func (s *serviceNetworkSynthesizer) synthesizeTriggeredGateways(ctx context.Cont
 	} else {
 		return nil
 	}
+}
+
+func (s *serviceNetworkSynthesizer) deleteServiceNetwork(ctx context.Context, resServiceNetwork *latticemodel.ServiceNetwork) string {
+	glog.V(6).Infof("Synthesizing Gateway: Del %v\n", resServiceNetwork.Spec.Name)
+
+	// TODO need to check if service network is referenced by gateway in other namespace
+	gwList := &gateway_api.GatewayList{}
+	s.Client.List(context.TODO(), gwList)
```

Comment:
nit: use context from function argument - ctx, not context.TODO()

Reply:
This was a straight refactor-to-method, but happy to update when I see these.